plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Utils.hs

```diff
@@ -109,25 +110,25 @@ an under-approximation of how many arguments the term may need.
 e.g. consider the term @let id = \x -> x in id@: the variable @id@ has syntactic
 arity @[]@, but does in fact need an argument before it does any work.
 -}
-type Arity = [ParamKind]
+type Arity tyname name = [Param tyname name]
 
 -- | Info attached to a let-binding needed for call site inlining.
-data VarInfo tyname name uni fun ann =
-  MkVarInfo {
-    varStrictness :: Strictness
-    ,varDef       :: InlineTerm tyname name uni fun ann
+data VarInfo tyname name uni fun ann = MkVarInfo
+    { varStrictness :: Strictness
+    , varDef        :: Term tyname name uni fun ann
```

Comment:
This needs to be an `InlineTerm`, see [Dupable](https://github.com/input-output-hk/plutus/blob/2d0a953ffcdce0975595c3c9422ff1202d7ae63e/plutus-core/plutus-core/src/PlutusCore/Rename/Internal.hs#L46) and [this comment](https://github.com/input-output-hk/plutus/pull/5251#discussion_r1163900145).

Reply:
Yes, this is suspicious.